src/cmd/flux-job.c

```diff
@@ -3355,6 +3385,158 @@ int cmd_purge (optparse_t *p, int argc, char **argv)
     return rc;
 }
 
+static struct taskmap *flux_job_taskmap (flux_jobid_t id)
+{
+    struct taskmap *map;
+    flux_t *h;
+    flux_future_t *f;
+
+    if (!(h = flux_open (NULL, 0)))
+        log_err_exit ("flux_open");
+
+    if (!(f = flux_job_event_watch (h, id, "guest.exec.eventlog", 0)))
+        log_err_exit ("flux_job_event_watch");
+    while (true) {
+        json_t *o;
+        json_t *context;
+        const char *event;
+        const char *name;
+        if (flux_job_event_watch_get (f, &event) < 0) {
+            if (errno == ENODATA)
+                log_msg_exit ("No taskmap found for job");
+            if (errno == ENOENT)
+                log_msg_exit ("Unable to get job taskmap: no such job");
+            log_msg_exit ("waiting for shell.start event: %s",
+                          future_strerror (f, errno));
+        }
+        if (!(o = eventlog_entry_decode (event)))
+            log_err_exit ("eventlog_entry_decode");
+        if (eventlog_entry_parse (o, NULL, &name, &context) < 0)
+            log_err_exit ("eventlog_entry_parse");
+        if (streq (name, "shell.start")) {
+            flux_error_t error;
+            if (!(o = json_object_get (context, "taskmap"))
+                || !(map = taskmap_decode_json (o, &error)))
+                log_msg_exit ("failed to get taskmap from shell.start event");
+            break;
+        }
+        flux_future_reset (f);
```

Comment:
The `o` from  `eventlog_entry_decode()` is overwritten in the `shell.start` block and leaked.